### PR TITLE
Unit test for free trust lines (RIPD-911)

### PR DESCRIPTION
**Pull request by bachase**
Unit test to ensure two free trust lines can be added for an account without extra reserve.

### CODE REVIEW

**Comment by mellery451** [comment]
👍

**Comment by vinniefalco** [comment]
Coverage went down?

File: .gitignore [hunk 1]
```diff
@@ -93,3 +93,4 @@ Builds/VisualStudio2015/*.sdf
 
 # MSVC
 *.pdb
+.vs/
```
Comment (vinniefalco) [inline .gitignore]: Missing carriage return

File: src/test/app/SetTrust_test.cpp [hunk 2]
```diff
@@ -0,0 +1,79 @@
+//------------------------------------------------------------------------------
+/*
+This file is part of rippled: https://github.com/ripple/rippled
```
Comment (vinniefalco) [inline src/test/app/SetTrust_test.cpp]: wrong indentation

File: src/unity/app_test_unity.cpp [hunk 3]
```diff
@@ -39,4 +39,5 @@
 #include <test/app/Taker_test.cpp>
 #include <test/app/Transaction_ordering_test.cpp>
 #include <test/app/TxQ_test.cpp>
-#include <test/app/ValidatorList_test.cpp>
\ No newline at end of file
+#include <test/app/ValidatorList_test.cpp>
+#include <test/app/SetTrust_test.cpp>
```
Comment (vinniefalco) [inline src/unity/app_test_unity.cpp]: missing carriage return

File: src/test/app/SetTrust_test.cpp [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+        env(trust(alice, gwC["USD"](100)), ter(tecNO_LINE_INSUF_RESERVE), require(lines(alice,2)));
+        env.close();
+
+        // Fund Alice additional amount to cover 
+        env(pay(env.master, alice, STAmount(threelineReserve-baseReserve)));
+        env.close();
+
+        env(trust(alice, gwC["USD"](100)), require(lines(alice,3)));
+        env.close();
+
```
Comment (vinniefalco) [inline src/test/app/SetTrust_test.cpp]: That's it?! Cool

**Comment by vinniefalco** [comment]
:+1: Great test

File: src/test/app/SetTrust_test.cpp [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include <ripple/test/jtx.h>
+
+namespace ripple {
+
```
Comment (miguelportilla) [inline src/test/app/SetTrust_test.cpp]: Nit pick - usually no linefeed here.

File: src/test/app/SetTrust_test.cpp [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+        testcase("Allow 2 free trust lines before requiring reserve");
+
+        using namespace jtx;
+        Env env(*this);
+
+        auto const gwA = Account{ "gatewayA" };
+        auto const gwB = Account{ "gatewayB" };
+        auto const gwC = Account{ "gatewayC" };
+        auto const alice = Account{ "alice" };
+
```
Comment (miguelportilla) [inline src/test/app/SetTrust_test.cpp]: Nit pick: extra space here.

I recommend you use a utility to automatically remove extra spaces when you save. If you use VS, this one is free and works well: https://visualstudiogallery.msdn.microsoft.com/a204e29b-1778-4dae-affd-209bea658a59

File: src/test/app/SetTrust_test.cpp [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+        env.fund(XRP(10000), gwA, gwB, gwC);
+
+        // Fund alice with ...
+        env.fund(baseReserve /* enough to hold an account */
+            + drops(3*txFee) /* and to pay for 3 transactions */, alice);
+        env.close();
+
+        env(trust(alice, gwA["USD"](100)), require(lines(alice,1)));
+        env.close();
+
```
Comment (miguelportilla) [inline src/test/app/SetTrust_test.cpp]: Nit pick: extra space here.

**Comment by miguelportilla** [comment]
:+1 @bachase Test looks great

**Comment by bachase** [comment]
Merged in b8cae2dfaf326b3e91dc5bad0066d2955d8ae3e8

**Pull request closed without merging**